src/server/auth.ts

```diff
@@ -53,3 +55,43 @@ export default async function initializePatronTokenAuth(req: NextRequest) {
 
   return patronTokenResponse
 }
+
+/**
+ * Creates and returns redirect url from initializePatronTokenAuth() response
+ */
+export function getLoginRedirect(req) {
+  if (
+    !req.patronTokenResponse ||
+    !req.patronTokenResponse.isTokenValid ||
+    !req.patronTokenResponse.decodedPatron ||
+    !req.patronTokenResponse.decodedPatron.sub
+  ) {
+    const protocol = "http"
+    const hostname = req.headers["host"]
+    const originalUrl = BASE_URL + req.url
+    const fullUrl = encodeURIComponent(
+      `${protocol}://${hostname}${originalUrl}`
+    )
+    const redirect =
+      //`${appConfig.externalUrls.login}?redirect_uri=${fullUrl}`
+      `${process.env.LOGIN_BASE_URL}/login?redirect_uri=${fullUrl}`
+    console.log(redirect)
+    return redirect
+  }
+}
+
+/**
+ * Creates redirect to log out user, then return user to their current page. Requires an instance of useRouter.
+ */
+
+export function getLogoutRedirect(router) {
+  const current = router.pathname
+  let backPath = current
+  // If the patron is on any hold or account page, then
+  // redirect them to the home page after logging out.
+  if (current.includes("hold") || current.includes("account")) {
+    backPath = "/"
+  }
+  return `${appConfig.externalUrls.logoutUrl}?redirect_uri=http://local.nypl.org:8080/research/research-catalog${backPath}`
+  // TODO: The home url needs to be an env variable fs
```

Comment:
DFE has this as  `${window.location.origin}`  instead of an env var but this function is not being called on the client side to update the logout url. I'd be curious about testing this because I'm not sure if hardcoding the home domain url will work in various instances (the prod aws instance and the reverse proxy instance).

This _does_ work though. We can chat about this.

Reply:
For now this works but update this to an env var before this goes to QA.